build/common/installer/scripts/tomlparser-agent-config.rb

```diff
@@ -57,6 +57,28 @@
 @fbitTailMemBufLimitMBs = 0
 @fbitTailIgnoreOlder = ""
 
+# Checking to see if this is the daemonset or replicaset to parse config accordingly
+@controllerType = ENV["CONTROLLER_TYPE"]
+@daemonset = "daemonset"
+# Checking to see if container is not prometheus sidecar.
+# CONTAINER_TYPE is populated only for prometheus sidecar container.
+@containerType = ENV["CONTAINER_TYPE"]
+
+@promFbitChunkSize = 0
+@promFbitBufferSize = 0
+@promFbitMemBufLimit = 0
+
+# set defaults
+if !@controllerType.nil? && !@controllerType.empty? && @controllerType.strip.casecmp(@daemonset) == 0
+  @promFbitChunkSize = 32
+  @promFbitBufferSize = 64
+  @promFbitMemBufLimit = 5
+elsif !@controllerType.nil? && !@controllerType.empty? && @controllerType.strip.casecmp(@daemonset) != 0
+  @promFbitChunkSize = 32
+  @promFbitBufferSize = 64
```

Comment:
we don't need to worry about 5m vs. 10m. Lets have 32,64, 10 as default and get rid of this condition.